api/client/network.go

```diff
@@ -222,7 +224,7 @@ func consolidateIpam(subnets, ranges, gateways []string, auxaddrs map[string]str
 				return nil, fmt.Errorf("multiple overlapping subnet configuration is not supported")
 			}
 		}
-		iData[s] = &network.IPAMConfig{Subnet: s, AuxAddress: map[string]string{}}
+		iData[s] = &network.IPAMConfig{Subnet: s, AuxAddress: map[string]string{}, Options: options}
```

Comment:
I missed this in my previous comments.
The `-ipam-opt` is a generic list of options that is sent to the IPAM driver for a particular network. But when you include it as part of `IPAMConfig`, it becomes a field for every subnet. That doesn't seem right.
I think instead of touching the `IPAMConfig` structure, this should be part of `types.NetworkCreate` structure and pass it to libnetwork. WDYT ?